contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -1187,3 +1192,85 @@ babelfish_integrity_checker(PG_FUNCTION_ARGS)
 
 	PG_RETURN_BOOL(true);
 }
+
+Datum
+bigint_degrees(PG_FUNCTION_ARGS)
+{
+	int64	arg1 = PG_GETARG_INT64(0);
+	float8	result;
+	 
+	result = DatumGetFloat8(DirectFunctionCall1(degrees, Float8GetDatum((float8) arg1)));
+
+	if (result < 0)
+		result = ceil(result);
+	else
+		result = floor(result);
+
+	 /* Range check */
+	if (unlikely(isnan(result) || !FLOAT8_FITS_IN_INT64(result)))
+		ereport(ERROR,
+				(errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
+				errmsg("Arithmetic overflow error converting expression to data type bigint")));
+
+	PG_RETURN_INT64((int64)result);
+}
+
+Datum
+int_degrees(PG_FUNCTION_ARGS)
+{
+	int32	arg1 = PG_GETARG_INT32(0);
+	float8	result;
+	 
+	result = DatumGetFloat8(DirectFunctionCall1(degrees, Float8GetDatum((float8) arg1)));
+
+	if (result < 0)
+		result = ceil(result);
+	else
+		result = floor(result);
+
+	 /* Range check */
+	if (unlikely(isnan(result) || !FLOAT8_FITS_IN_INT32(result)))
```

Comment:
Do we have a test case that can exercise this code?

Reply:
SELECT degrees(CAST(2147483648 AS INT));

Reply:
The above example throws an error in `CAST(2147483648 AS INT)` part, not in this function.